[MM-485] - Update `commonHelpText` and `sysAdminHelpText` to include all slash commands

Francois-D opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary

*Problem*
jiraCommandHandler contains the mappings of slash commands to their corresponding command handler functions. The following slash commands are missing and should be added to the HelpText variables.

*Resolution*
Mapped the following commands with their descriptions to the handler functions. 
webhook
stats
info
help

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-plugin-jira/issues/485

jfrerich commented on server/command.go:
```diff
@@ -21,6 +21,10 @@ const commonHelpText = "\n* `/jira connect` - Connect your Mattermost account to
 	"* `/jira unassign <issue-key>` - Unassign the Jira issue\n" +
 	"* `/jira create <text (optional)>` - Create a new Issue with 'text' inserted into the description field\n" +
 	"* `/jira transition <issue-key> <state>` - Change the state of a Jira issue\n" +
+	"* `/jira stats` - Poll basic statistics about the Jira plug-in.  Recevie performace metrics and authenticated users between Jira/Mattermost\n" +
```
```suggestion
	"* `/jira stats` - Poll basic statistics about the Jira plug-in.  Receive performance metrics and authenticated users between Jira/Mattermost\n" +
```

jfrerich commented on server/command.go:
```diff
@@ -21,6 +21,10 @@ const commonHelpText = "\n* `/jira connect` - Connect your Mattermost account to
 	"* `/jira transition <issue-key> <state>` - Change the state of a Jira issue\n" +
+	"* `/jira stats` - Poll basic statistics about the Jira plug-in.  Recevie performace metrics and authenticated users between Jira/Mattermost\n" +
+	"* `/jira info` - Retrieve information about the current user and the Jira plug-in\n" +
+	"* `/jira help` - Launch the Jira plug-in command line help syntax\n" +
```
```suggestion
	"* `/jira help` - Launch the Jira plugin command line help syntax\n" +
```

jfrerich commented on server/command.go:
```diff
@@ -21,6 +21,10 @@ const commonHelpText = "\n* `/jira connect` - Connect your Mattermost account to
+	"* `/jira stats` - Poll basic statistics about the Jira plug-in.  Recevie performace metrics and authenticated users between Jira/Mattermost\n" +
+	"* `/jira info` - Retrieve information about the current user and the Jira plug-in\n" +
+	"* `/jira help` - Launch the Jira plug-in command line help syntax\n" +
+	"* `/jira webhook` - Execute the Jira plug-in webhook. The Jira plug-in gets sent a stream of pre-configured events from the Jira server\n" +
```
```suggestion
	"* `/jira webhook` - Execute the Jira plug-in webhook. The Jira plugin gets sent a stream of pre-configured events from the Jira server\n" +
```

Francois-D commented:
@jfrerich sorry about the typos.
I've commited my edits.

levb commented on server/command.go:
```diff
@@ -698,7 +702,7 @@ func getCommand() *model.Command {
 		Description:      "Integration with Jira.",
 		AutoComplete:     true,
-		AutoCompleteDesc: "Available commands: connect, assign, disconnect, create, transition, view, subscribe, settings, install cloud/server, uninstall cloud/server, help",
+		AutoCompleteDesc: "Available commands: connect, assign, disconnect, create, transition, stats, info, webhook,  view, subscribe, settings, install cloud/server, uninstall cloud/server, help",
```
remove stats

Francois-D replied:
The following commands are also listed as system administrator commands should they be removed as well including 'stats'?
_Install cloud/server, webhook, subscribe._

levb replied:
Yes, thanks for catching these.

Francois-D replied:
Thanks I've made the edits. :)

levb replied:
@Francois-D also, these descriptions belong in the SysAdmin-specific section, just a few lines below

Francois-D commented on server/command.go:
```diff
@@ -21,6 +21,10 @@ const commonHelpText = "\n* `/jira connect` - Connect your Mattermost account to
+	"* `/jira stats` - Poll basic statistics about the Jira plug-in.  Receive performance metrics and authenticated users between Jira/Mattermost\n" +
+	"* `/jira info` - Retrieve information about the current user and the Jira plug-in\n" +
+	"* `/jira help` - Launch the Jira plugin command line help syntax\n" +
+	"* `/jira webhook` - Execute the Jira plug-in webhook. The Jira plugin gets sent a stream of pre-configured events from the Jira server\n" +
```
Thanks for the feedback.  The Jira stat description was derived from this location.  
https://mattermost.gitbook.io/plugin-jira/admininstrator-guide/administrator-slash-commands#retrieving-basic-stats

Is this the same slash command that Jira stats is referring to?  
Thanks in advance

levb replied:
`/jira webhook` should be "Display the URL to configure Jira webhooks to".

levb commented on server/command.go:
```diff
@@ -21,6 +21,10 @@ const commonHelpText = "\n* `/jira connect` - Connect your Mattermost account to
 	"* `/jira unassign <issue-key>` - Unassign the Jira issue\n" +
 	"* `/jira create <text (optional)>` - Create a new Issue with 'text' inserted into the description field\n" +
 	"* `/jira transition <issue-key> <state>` - Change the state of a Jira issue\n" +
+	"* `/jira stats` - Retrieve version information about the Jira plugin\n" +
```
2/5 suggest: "Display usage statistics"

Francois-D replied:
Thanks.
I've updated description.

levb commented on server/command.go:
```diff
@@ -21,6 +21,10 @@ const commonHelpText = "\n* `/jira connect` - Connect your Mattermost account to
 	"* `/jira create <text (optional)>` - Create a new Issue with 'text' inserted into the description field\n" +
 	"* `/jira transition <issue-key> <state>` - Change the state of a Jira issue\n" +
+	"* `/jira stats` - Retrieve version information about the Jira plugin\n" +
+	"* `/jira info` - Retrieve information about the current user and the Jira plug-in\n" +
```
1/5 `s/Retrieve/Display/`

levb commented:
@Francois-D also, these descriptions belong in the SysAdmin-specific section, just a few lines below

Francois-D commented:
> @Francois-D also, these descriptions belong in the SysAdmin-specific section, just a few lines below

Thanks I've moved the command to the correct area.  I also updated the subscribe list description with the official one from the documentation from the link 

https://mattermost.gitbook.io/plugin-jira/admininstrator-guide/administrator-slash-commands#jira-subscribe-list-command

DHaussermann commented:
@Francois-D Thanks for updating this!
One small point about some of the new text - for `/jira webhook` the new text is "_Configure the Mattermost webhook to receive JQL queries_" 
As far as I know, there is no way to change any of the configuration from this command. Would it maybe make more sense to just have `/jira webhook` the new text is "_**Show** the Mattermost webhook to receive JQL queries_"

Francois-D commented:
@DHaussermann 
Thanks I've updated the WebHooks definition.

Francois-D commented:
@DHaussermann 
May I suggest informing the user to run /jira webhook? This would aid in setting up the plugin.

<img width="1201" alt="Screen Shot 2020-04-28 at 1 49 04 PM" src="https://user-images.githubusercontent.com/46973576/80521185-9a797580-8958-11ea-8e04-ea9842cac652.png">


Example of running /jira webhook
<img width="820" alt="Screen Shot 2020-04-28 at 1 56 40 PM" src="https://user-images.githubusercontent.com/46973576/80520938-2e970d00-8958-11ea-8a12-be06d0c02065.png">
